cpp/src/reductions/scan/ewm.cu

```diff
@@ -0,0 +1,384 @@
+
+    thrust::transform(
+      rmm::exec_policy(stream),
+      pairs.begin(),
+      pairs.end(),
+      output.begin(),
+      output.begin(),
+      [] __device__(pair_type<T> pair, T numerator) -> T { return numerator / pair.second; });
+
+  } else {
+    // Numerator
+    // Fill with pairs
+    thrust::transform(rmm::exec_policy(stream),
+                      input.begin<T>(),
+                      input.end<T>(),
+                      pairs.begin(),
+                      [beta] __device__(T input) -> pair_type<T> {
+                        return {beta, input};
+                      });
+    compute_recurrence(pairs, stream);
+
+    // copy the second elements to the output for now
+    thrust::transform(rmm::exec_policy(stream),
+                      pairs.begin(),
+                      pairs.end(),
+                      output.begin(),
+                      [] __device__(pair_type<T> pair) -> T { return pair.second; });
+
+    // Denominator
+    // Fill with pairs
+    thrust::fill(rmm::exec_policy(stream), pairs.begin(), pairs.end(), pair_type<T>(beta, 1.0));
+    compute_recurrence(pairs, stream);
+
+    thrust::transform(
+      rmm::exec_policy(stream),
+      pairs.begin(),
+      pairs.end(),
+      output.begin(),
+      output.begin(),
+      [] __device__(pair_type<T> pair, T numerator) -> T { return numerator / pair.second; });
+  }
+  return output;
+}
+
+template <typename T>
+rmm::device_uvector<T> compute_ewma_noadjust(column_view const& input,
+                                             T const beta,
+                                             rmm::cuda_stream_view stream,
+                                             rmm::mr::device_memory_resource* mr)
+{
+  rmm::device_uvector<T> output(input.size(), stream);
+  rmm::device_uvector<pair_type<T>> pairs(input.size(), stream);
+
+  thrust::transform(rmm::exec_policy(stream),
+                    input.begin<T>(),
+                    input.end<T>(),
+                    pairs.begin(),
+                    [beta] __device__(T input) -> pair_type<T> {
+                      return {beta, (1.0 - beta) * input};
+                    });
+
+  // TODO: the first pair is WRONG using the above. Reset just that pair
+  thrust::transform(rmm::exec_policy(stream),
+                    input.begin<T>(),
+                    std::next(input.begin<T>()),
+                    pairs.begin(),
+                    [beta] __device__(T input) -> pair_type<T> {
+                      return {beta, input};
+                    });
+
+  if (input.has_nulls()) {
+    /*
+    In this case, a denominator actually has to be computed. The formula is
+    y_{i+1} = (1 - alpha)x_{i-1} + alpha x_i, but really there is a "denominator"
+    which is the sum of the weights: alpha + (1 - alpha) == 1. If a null is
+    encountered, that means that the "previous" value is downweighted by a
+    factor (for each missing value). For example this would y_2 be for one NULL:
```

Comment:
You're not capitalizing null anywhere else that I saw, no reason to start now.
```suggestion
    factor (for each missing value). For example this would y_2 be for one null:
```